frontend/device/android/powerd.lua

```diff
@@ -69,6 +74,12 @@ function AndroidPowerD:turnOnFrontlightHW(done_callback)
 
     android.setScreenBrightness(math.floor(self.fl_intensity * self.bright_diff / self.fl_max))
 
+    -- Devices using OnyxWarmthController also need to turn on warmth
+    if android.hasStandaloneWarmth() then
+        -- Due to AndroidPowerD:frontlightWarmthHW() above, we need to divide by self.warm_diff,
```

Comment:
no comment, please.

If you find hard to understand what's happening it is better to refactor the bit of code that made you doubt on the first place.

Like above, the comment isn't neccesary nor 100% accurate.

Reply:
On a slightly related matter, re: the `warm_diff` stuff

I don't have any device to test, but, FWIW, PowerD now provides `fromNativeWarmth`/`toNativeWarmth` methods, that handle the insanity of the public warmth API being stuck with a [0, 100] range despite basically almost no devices doing that ;p.

(The scale is computed on Generic's init, based on `fl_warmth_max`; c.f., Kobo for funkier setups that need it earlier).

Reply:
> PowerD now provides `fromNativeWarmth`/`toNativeWarmth` methods

I don't think this would help here, because the problem was not that ranges were mismatched (like [0-100] vs [0-255]), but in the fact that `fl_warmth` was multiplied by `warm_diff` _twice_.

I guess the whole thing needs a careful refactor, but I don't have any other devices to test, and I put all these changes in the OnyxWarmthController-guarded `if` block :D 

If you guys agree that this works, we can merge it, and then take another steps to fine tune it.

(Also, I won't have physical access to the device to test again for the couple of days).

Reply:
> > PowerD now provides `fromNativeWarmth`/`toNativeWarmth` methods
> 
> I don't think this would help here, because the problem was not that ranges were mismatched (like [0-100] vs [0-255]), but in the fact that `fl_warmth` was multiplied by `warm_diff` _twice_.
> 
> I guess the whole thing needs a careful refactor, but I don't have any other devices to test, and I put all these changes in the OnyxWarmthController-guarded `if` block :D
> 
> If you guys agree that this works, we can merge it, and then take another steps to fine tune it.
> 
> (Also, I won't have physical access to the device to test again for the couple of days).

I can test when I find some time. The values for new Onyx devices are even more bizarre: https://github.com/hugleo/android-luajit-launcher/blob/f9097de92f2517afa1ed60dc6b01fe85658c3f08/app/src/main/java/org/koreader/launcher/device/lights/OnyxAdbLightsController.kt#L13C1-L14